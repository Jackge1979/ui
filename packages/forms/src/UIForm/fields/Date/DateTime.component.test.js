import React from 'react';
import { shallow } from 'enzyme';
import DateTimeWidget from './DateTime.component';

describe('DateTime widget', () => {
	const schema = {
		autoFocus: true,
		description: 'talend datetime picker',
		placeholder: 'YYYY-MM-DD HH:mm',
		required: true,
		title: 'Select DateTime',
		type: 'text',
		schema: { type: 'string' },
	};
	it('should render a InputDateTimePicker', () => {
		// when
		const wrapper = shallow(
			<DateTimeWidget
				id="talend-date-time"
				isValid={false}
				errorMessage="something wrong"
				schema={schema}
				onChange={jest.fn()}
				onFinish={jest.fn()}
				options={{ dafeFormat: 'DD/MM/YYYY' }}
			/>,
		);

		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
