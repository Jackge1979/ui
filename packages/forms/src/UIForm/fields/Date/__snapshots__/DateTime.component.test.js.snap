// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTime widget should render a InputDateTimePicker 1`] = `
<FieldTemplate
  description="talend datetime picker"
  descriptionId="talend-date-time-description"
  errorId="talend-date-time-error"
  errorMessage="something wrong"
  id="talend-date-time"
  isValid={false}
  label="Select DateTime"
  required={true}
>
  <InputDateTimePicker
    aria-describedby="talend-date-time-description talend-date-time-error"
    aria-invalid={true}
    aria-required={true}
    autoFocus={true}
    dateFormat="YYYY-MM-DD"
    id="talend-date-time"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="YYYY-MM-DD HH:mm"
    required={true}
    useSeconds={false}
    useUTC={false}
  />
</FieldTemplate>
`;
